File: bin/spark/example-scala/spark-applications/NameList.py
import sys
from pyspark.sql import SparkSession

if __name__ == "__main__":
  if len(sys.argv) < 3:
    print >> sys.stderr, "Usage: NameList.py <input-data-source> <output-data-source>"
    sys.exit()

  spark = SparkSession.builder.getOrCreate()
  spark.sparkContext.setLogLevel("WARN")

  peopleDF = spark.read.json(sys.argv[1])
  namesDF = peopleDF.select("firstName","lastName")
  namesDF.write.option("header","true").csv(sys.argv[2])

  spark.stop()

File: bin/spark/example-scala/spark-applications/namelist_project/src/main/scala/example/NameList.scala
package example

import org.apache.spark.sql.SparkSession

object NameList {
  def main(args: Array[String]) {
    if (args.length < 2) {
      System.err.println("Usage: example.NameList <input-data-source> <output-data-source>")
      System.exit(1)
    }

    val spark = SparkSession.builder.getOrCreate()
    spark.sparkContext.setLogLevel("WARN")

    val peopleDF = spark.read.json(args(0))
    val namesDF = peopleDF.select("firstName","lastName")
    namesDF.write.option("header","true").csv(args(1))

    spark.stop()
  }
}

